packages/@aws-cdk/aws-ecs/lib/queue-worker-service-base.ts

```diff
@@ -0,0 +1,180 @@
+   * The image to start.
+   */
+  readonly image: ContainerImage;
+
+  /**
+   * The CMD value to pass to the container. A string with commands delimited by commas.
+   *
+   * @default none
+   */
+  readonly command?: string;
+
+  /**
+   * Number of desired copies of running tasks
+   *
+   * @default 1
+   */
+  readonly desiredTaskCount?: number;
+
+  /**
+   * Flag to indicate whether to enable logging
+   *
+   * @default true
+   */
+  readonly enableLogging?: boolean;
+
+  /**
+   * The environment variables to pass to the container.
+   *
+   * @default none
+   */
+  readonly environment?: { [key: string]: string };
+
+  /**
+   * A name for the queue.
+   *
+   * If specified and this is a FIFO queue, must end in the string '.fifo'.
+   *
+   * @default 'EcsWorkerServiceQueue'
+   */
+  readonly queueName?: string;
+
+  /**
+   * Maximum capacity to scale to.
+   *
+   * @default (desiredTaskCount * 2)
+   */
+  readonly maxScalingCapacity?: number
+
+  /**
+   * The intervals for scaling based on the SQS queue's ApproximateNumberOfMessagesVisible metric.
+   *
+   * Maps a range of metric values to a particular scaling behavior.
+   * https://docs.aws.amazon.com/autoscaling/ec2/userguide/as-scaling-simple-step.html
+   *
+   * @default [{ upper: 0, change: -1 },{ lower: 100, change: +1 },{ lower: 500, change: +5 }]
+   */
+  readonly scalingSteps?: autoscaling.ScalingInterval[];
+}
+
+/**
+ * Base class for a Fargate and ECS queue worker service
+ */
+export abstract class QueueWorkerServiceBase extends cdk.Construct {
+  /**
+   * The SQS queue that the worker service will process from
+   */
+  public readonly sqsQueue: Queue;
+
+  // Properties that have defaults defined. The Queue Worker will handle assigning undefined properties with default
+  // values so that derived classes do not need to maintain the same logic.
+
+  /**
+   * Environment variables that will include the queue name
+   */
+  public readonly environment: { [key: string]: string };
+  /**
+   * The minimum number of tasks to run
+   */
+  public readonly desiredCount: number;
+  /**
+   * The maximum number of instances for autoscaling to scale up to
+   */
+  public readonly maxCapacity: number;
+  /**
+   * The scaling interval for autoscaling based off an SQS Queue size
+   */
+  public readonly scalingSteps: autoscaling.ScalingInterval[];
+  /**
+   * The AwsLogDriver to use for logging if logging is enabled.
+   */
+  public readonly logDriver?: LogDriver;
+
+  constructor(scope: cdk.Construct, id: string, props: QueueWorkerServiceBaseProps) {
+    super(scope, id);
+
+    // Create the worker SQS queue
+    const queueName = props.queueName !== undefined ? props.queueName : 'EcsWorkerServiceQueue';
+    this.sqsQueue = new sqs.Queue(this, 'EcsWorkerServiceQueue', {
+      queueName
+    });
+
+    // Setup autoscaling scaling intervals
+    const defaultScalingSteps = [{ upper: 0, change: -1 }, { lower: 100, change: +1 }, { lower: 500, change: +5 }];
```

Comment:
Would be curious to hear how you came up with these numbers.

Reply:
This was discussed in the design: https://github.com/awslabs/aws-cdk/pull/2402. After discussion with @clareliguori and @nathanpeck, we agreed that these would be reasonable defaults and the user can override the default if it's different for their use case.